app/controllers/v0/facilities/ccp_controller.rb

```diff
@@ -36,6 +29,37 @@ def services
 
   private
 
+  def ppms_search
+    if Flipper.enabled?(:facility_locator_ppms_legacy_urgent_care_to_pos_locator)
+      ppms_search_with_legacy_urgent_care
+    else
+      ppms_search_without_legacy_urgent_care
+    end
+  end
+
+  def ppms_search_without_legacy_urgent_care
+    case search_params[:type]
+    when 'cc_provider'
+      provider_search
+    when 'cc_pharmacy'
+      provider_search('services' => ['3336C0003X'])
+    when 'cc_urgent_care'
+      api.pos_locator(search_params)
+    end
+  end
+
+  def ppms_search_with_legacy_urgent_care
+    if search_params[:type] == 'cc_provider' && search_params['services'] == ['261QU0200X']
```

Comment:
minor optimization: you could put the `search_params['services'] == ['261QU0200X']` predicate first so you don't have to evaluate `search_params[:type] == 'cc_provider'` twice when it's true

Reply:
I can't, `type` has precedence. so if they pass cc_pharmacy I don't want them to get a POS search

Reply:
sorry, I meant reversing them in this line like:
```
search_params['services'] == ['261QU0200X'] && search_params[:type] == 'cc_provider' 
```
so when `search_params['services'] == ['261QU0200X']` is false it won't immediately check `search_params[:type] == 'cc_provider'` again on the next line

Reply:
I think the performance hit on this is minor. we already have the params hash. I think your proposed change makes it more difficult to read. I think it was breaking the complexity checker but originally it was a case statement with an if  inside the `cc_provider` case.